presto-main/src/main/java/io/prestosql/sql/ExpressionUtils.java

```diff
@@ -330,21 +327,6 @@ else if (!seen.contains(expression)) {
         return result.build();
     }
 
-    public static Expression normalize(Expression expression)
-    {
-        if (expression instanceof NotExpression) {
```

Comment:
Could we keep normalization for non doubles? `Normalization` is only called from planner, so types should be available.

Concern here is that automated tools would generate odd expressions (e.g `not(not(...))` or `not(col1_date > col2_date)`. Such not normalized expressions would prevent optimized inequality join from working.

Reply:
This is already handle by the optimizer (PushDownNegationsExpressionRewriter), so we don't lose any optimizations